File: examples/python/basic.py
"""
Basic ensureJson usage examples.

This example demonstrates the core functionality of ensureJson.
"""

from ensure_json import ensure_json, JsonFixError


def example_1_markdown_fences():
    """Remove markdown code fences."""
    print("=" * 60)
    print("Example 1: Markdown Fences")
    print("=" * 60)

    llm_output = '''
Here's your data:
```json
{
  "name": "Alice",
  "age": 30
}
```
    '''

    result = ensure_json(llm_output)
    print(f"Input: {llm_output[:50]}...")
    print(f"Output: {result}")
    print()


def example_2_trailing_commas():
    """Fix trailing commas."""
    print("=" * 60)
    print("Example 2: Trailing Commas")
    print("=" * 60)

    broken = '{"name": "Bob", "hobbies": ["reading", "coding",],}'

    result = ensure_json(broken)
    print(f"Input: {broken}")
    print(f"Output: {result}")
    print()


def example_3_unquoted_keys():
    """Quote bare keys."""
    print("=" * 60)
    print("Example 3: Unquoted Keys")
    print("=" * 60)

    broken = '{name: "Charlie", age: 25, city: "NYC"}'

    result = ensure_json(broken)
    print(f"Input: {broken}")
    print(f"Output: {result}")
    print()


def example_4_mixed_quotes():
    """Handle mixed quote styles."""
    print("=" * 60)
    print("Example 4: Mixed Quotes")
    print("=" * 60)

    broken = "{name: 'Diana', 'age': 28, \"city\": 'London'}"

    result = ensure_json(broken)
    print(f"Input: {broken}")
    print(f"Output: {result}")
    print()


def example_5_incomplete_json():
    """Balance missing brackets."""
    print("=" * 60)
    print("Example 5: Incomplete JSON")
    print("=" * 60)

    broken = '{"users": [{"name": "Eve"}, {"name": "Frank"'

    result = ensure_json(broken)
    print(f"Input: {broken}")
    print(f"Output: {result}")
    print()


def example_6_complex_llm_output():
    """Real-world LLM output."""
    print("=" * 60)
    print("Example 6: Complex LLM Output")
    print("=" * 60)

    llm_output = '''
I'll provide the user data in JSON format:

```json
{
  name: "Grace Hopper",
  occupation: 'Computer Scientist',
  achievements: [
    "COBOL inventor",
    "First compiler",
    "Debugging pioneer",
  ],
  born: 1906,
}
```

Hope that helps!
    '''

    result = ensure_json(llm_output)
    print(f"Input: {llm_output[:80]}...")
    print(f"Output: {result}")
    print()


def example_7_error_handling():
    """Demonstrate error handling."""
    print("=" * 60)
    print("Example 7: Error Handling")
    print("=" * 60)

    invalid_input = "This is not JSON at all!"

    try:
        result = ensure_json(invalid_input)
        print(f"Result: {result}")
    except JsonFixError as err:
        print(f"❌ Error: {err}")
        print(f"   Original input: {err.raw}")
    print()


if __name__ == "__main__":
    print("\n🔧 ensureJson - Basic Examples\n")

    example_1_markdown_fences()
    example_2_trailing_commas()
    example_3_unquoted_keys()
    example_4_mixed_quotes()
    example_5_incomplete_json()
    example_6_complex_llm_output()
    example_7_error_handling()

    print("✅ All examples completed!")

File: examples/javascript/basic.js
/**
 * Basic ensureJson usage examples (JavaScript).
 *
 * This example demonstrates the core functionality of ensureJson.
 *
 * Usage:
 *   npm install ensure-json  # or pnpm add ensure-json
 *   node basic.js
 */

const { ensureJson, JsonFixError } = require('ensure-json');


function example1MarkdownFences() {
  console.log("=".repeat(60));
  console.log("Example 1: Markdown Fences");
  console.log("=".repeat(60));

  const llmOutput = `
Here's your data:
\`\`\`json
{
  "name": "Alice",
  "age": 30
}
\`\`\`
  `;

  const result = ensureJson(llmOutput);
  console.log(`Input: ${llmOutput.substring(0, 50)}...`);
  console.log(`Output:`, result);
  console.log();
}


function example2TrailingCommas() {
  console.log("=".repeat(60));
  console.log("Example 2: Trailing Commas");
  console.log("=".repeat(60));

  const broken = '{"name": "Bob", "hobbies": ["reading", "coding",],}';

  const result = ensureJson(broken);
  console.log(`Input: ${broken}`);
  console.log(`Output:`, result);
  console.log();
}


function example3UnquotedKeys() {
  console.log("=".repeat(60));
  console.log("Example 3: Unquoted Keys");
  console.log("=".repeat(60));

  const broken = '{name: "Charlie", age: 25, city: "NYC"}';

  const result = ensureJson(broken);
  console.log(`Input: ${broken}`);
  console.log(`Output:`, result);
  console.log();
}


function example4MixedQuotes() {
  console.log("=".repeat(60));
  console.log("Example 4: Mixed Quotes");
  console.log("=".repeat(60));

  const broken = "{name: 'Diana', 'age': 28, \"city\": 'London'}";

  const result = ensureJson(broken);
  console.log(`Input: ${broken}`);
  console.log(`Output:`, result);
  console.log();
}


function example5IncompleteJson() {
  console.log("=".repeat(60));
  console.log("Example 5: Incomplete JSON");
  console.log("=".repeat(60));

  const broken = '{"users": [{"name": "Eve"}, {"name": "Frank"';

  const result = ensureJson(broken);
  console.log(`Input: ${broken}`);
  console.log(`Output:`, result);
  console.log();
}


function example6ComplexLlmOutput() {
  console.log("=".repeat(60));
  console.log("Example 6: Complex LLM Output");
  console.log("=".repeat(60));

  const llmOutput = `
I'll provide the user data in JSON format:

\`\`\`json
{
  name: "Grace Hopper",
  occupation: 'Computer Scientist',
  achievements: [
    "COBOL inventor",
    "First compiler",
    "Debugging pioneer",
  ],
  born: 1906,
}
\`\`\`

Hope that helps!
  `;

  const result = ensureJson(llmOutput);
  console.log(`Input: ${llmOutput.substring(0, 80)}...`);
  console.log(`Output:`, result);
  console.log();
}


function example7ErrorHandling() {
  console.log("=".repeat(60));
  console.log("Example 7: Error Handling");
  console.log("=".repeat(60));

  const invalidInput = "This is not JSON at all!";

  try {
    const result = ensureJson(invalidInput);
    console.log(`Result:`, result);
  } catch (err) {
    if (err instanceof JsonFixError) {
      console.log(`❌ Error: ${err.message}`);
      console.log(`   Original input: ${err.raw}`);
    }
  }
  console.log();
}


function main() {
  console.log("\n🔧 ensureJson - Basic Examples (JavaScript)\n");

  example1MarkdownFences();
  example2TrailingCommas();
  example3UnquotedKeys();
  example4MixedQuotes();
  example5IncompleteJson();
  example6ComplexLlmOutput();
  example7ErrorHandling();

  console.log("✅ All examples completed!");
}


main();
